pkg/util/fast_int_set.go

```diff
@@ -64,25 +75,108 @@ func (s *FastIntSet) Empty() bool {
 	return s.smallVals == 0 && len(s.largeVals) == 0
 }
 
+// Next returns the first value in the set which is >= startVal. If there is no
+// value, the second return value is false.
+// Note: this is efficient for small sets, but each call takes linear time for large sets.
+func (s *FastIntSet) Next(startVal uint32) (uint32, bool) {
+	if startVal < smallValCutoff {
+		for v := s.smallVals >> startVal; v > 0; {
```

Comment:
On `bits`, yeah, I have a TODO somewhere to switch to those.

Thanks for the pointer on `Sparse`, that is very useful! I will look at switching to that.